spec/features/authentication_spec.rb

```diff
@@ -32,6 +32,47 @@
         expect(page).to have_content(I18n.t('sessions.create.failure'))
       end
     end
+
+    context 'with no authentication token' do
+      let(:request_for_comment) { create(:rfc_with_comment, user: user) }
+
+      it 'denies access to the request for comment' do
+        visit(request_for_comment_path(request_for_comment))
+        expect(page).to have_content(I18n.t('application.not_authorized'))
+      end
+    end
+
+    context 'with an authentication token' do
+      let(:user) { create(:learner) }
+      let(:request_for_comment) { create(:rfc_with_comment, user: user) }
+      let(:commenting_user) { InternalUser.create(attributes_for(:teacher)) }
+      let(:mail) { UserMailer.got_new_comment(request_for_comment.comments.first, request_for_comment, commenting_user) }
+      let(:rfc_link) { request_for_comment_url(request_for_comment, token: token.shared_secret) }
+
+      before { allow(AuthenticationToken).to receive(:generate!).with(user).and_return(token).once }
+
+      context 'when the token is valid' do
+        let(:token) { create(:authentication_token, user: user) }
+
+        it 'allows access to the request for comment' do
+          mail.deliver_now
+          visit(rfc_link)
+          expect(page).to have_content(request_for_comment.exercise.title)
```

Comment:
We could check here for the URL as well:
```
expect(current_url).to be(rfc_link)
```

Reply:
That's rather a case for 

```suggestion
          expect(response).to have_http_status :ok
          expect(page).to have_content(request_for_comment.exercise.title)
```

Reply:
As mentioned above, using a response in feature specs is neither supported nor recommended (as this should be done in a request spec). Therefore, we could not implement the spec as suggested here, but rather sticked with the existing method to verify the URL and the exercise title.